core/src/main/scala/org/apache/spark/sql/delta/stats/PrepareDeltaScan.scala

```diff
@@ -233,6 +215,7 @@ trait PrepareDeltaScanBase extends Rule[LogicalPlan]
       // It will fall back to just partition pruning at planning time.
       plan
     }
+    preprocessTablesWithDVs(updatedPlan)
```

Comment:
QQ: How does this handle the `V2WriteCommand` case?

Reply:
I am assuming you are talking about the case when stats are disabled?

I think there are existing issues around when stats are disabled. Planning to look at it once this DV project is complete. Currently when stats are disabled, we create `TahoeLogFileIndex` which doesn't have a pinned snapshot. Also the subquery handling part is missing.

Reply:
NVM. I didn't notice we returned directly when hitting `V2WriteCommand`.